storm-core/src/clj/org/apache/storm/stats.clj

```diff
@@ -188,14 +168,12 @@
 
 (defn value-bolt-stats!
   [^BoltExecutorStats stats]
-  (cleanup-bolt-stats! stats)
   (merge (value-common-stats (:common stats))
          (value-stats stats BOLT-FIELDS)
          {:type :bolt}))
 
 (defn value-spout-stats!
   [^SpoutExecutorStats stats]
-  (cleanup-spout-stats! stats)
```

Comment:
Really the issue is just that cleanup-bolt-stats! and cleanup-spout-stats! needs to move from value-*-stats! to something that is called from the executor shutdown.  

https://github.com/abhishekagarwal87/storm/blob/STORM-1693/storm-core/src/clj/org/apache/storm/daemon/executor.clj#L409-L417

I thought that is where they were called from when I saw them, but I obviously was wrong.